extensions/wikia/IndexingPipeline/PipelineRoutingBuilder.class.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Wikia\IndexingPipeline;
+
+class PipelineRoutingBuilder {
+	const NS_CONTENT = 'content';
+	const ROUTE_ACTION_KEY = '_action';
+	const ROUTE_NAMESPACE_KEY = '_namespace';
+	const ROUTE_TYPE_KEY = '_type';
+	const ACTION_CREATE = 'create';
+	const ACTION_UPDATE = 'update';
+	const ACTION_DELETE = 'delete';
+	const TYPE_ARTICLE = 'article';
+	const TYPE_WIKIA = 'wikia';
+
+	protected $keys;
+	protected $name;
+
+	protected function __construct() {
+		$this->keys = [ ];
+	}
+
+	/**
+	 * Provides routing key builder
+	 * @return PipelineRoutingBuilder
+	 */
+	public static function create() {
+		return new PipelineRoutingBuilder();
+	}
+
+	/**
+	 * Sets producer name
+	 *
+	 * @param string $name
+	 *
+	 * @return PipelineRoutingBuilder $this
+	 */
+	public function addName( $name ) {
+		$this->name = $name;
+
+		return $this;
+	}
+
+	public function addType( $type ) {
+		if ( in_array( $type, [ self::TYPE_ARTICLE, self::TYPE_WIKIA ] ) ) {
```

Comment:
I think t'd make sense to extract the condition and do if ($this->isSupportedType( $type ) )